packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,540 @@
+/// of the [CupertinoTableView], displays titles for each section in the
+/// [CupertinoTableView], and allows users to tap on a given section title to
+/// automatically scroll to that section.
+///
+/// See also:
+///
+///  * [CupertinoTableViewCell], which renders an iOS-style table view cell that
+///    is intended to appear within a [CupertinoTableView].
+///  * [ListView], which renders a style agnostic list.
+class CupertinoTableView extends StatefulWidget {
+  /// Creates a plain-style [CupertinoTableView] with an explicit list of cells.
+  const CupertinoTableView.plain({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildren = const <Widget>[],
+  }) : plainChildrenBuilder = null,
+       plainChildCount = null;
+
+  /// Creates a plain-style [CupertinoTableView] with a lazily built list of cells.
+  const CupertinoTableView.plainBuilder({
+    this.scrollController,
+    this.backgroundColor = CupertinoColors.white,
+    this.plainChildCount,
+    @required this.plainChildrenBuilder,
+  }) : plainChildren = null;
+
+  /// ScrollController that controls the scroll position of this [CupertinoTableView].
+  final ScrollController scrollController;
+
+  /// The table's background color beneath its cells.
+  final Color backgroundColor;
+
+  /// Explicit list of child cell widgets. If [plainChildren] is provided then
+  /// [plainChildrenBuilder] must be null.
+  final List<Widget> plainChildren;
+
+  /// Lazily built list of child cell widgets. If [plainChildrenBuilder] is
+  /// provided then [plainChildren] must be null.
+  final IndexedWidgetBuilder plainChildrenBuilder;
+
+  /// The number of children that can be built with [plainChildrenBuilder].
+  ///
+  /// The [plainChildCount] is optional. If it's provided then only a subset
+  /// of the plain children will be built at a time, based on viewport size. If
+  /// [plainChildCount] is not provided, then all plain children will be
+  /// created from the very beginning.
+  final int plainChildCount;
+
+  @override
+  _CupertinoTableViewState createState() => _CupertinoTableViewState();
+}
+
+class _CupertinoTableViewState extends State<CupertinoTableView> {
+  // Controller that reports the size of the last cell in the table, used to
+  // determine the desired gap between all extra dividers below it.
+  final _CellHeightController _cellHeightController = _CellHeightController();
+
+  // Decorator that builds cell widgets with extra chrome and/or functionality,
+  // e.g., adding a divider to a cell.
+  IndexedWidgetBuilder _cellDecorator;
+
+  // Cache of undecorated cell widgets. The cache is cleared every time this
+  // widget runs `build()`. The purpose of this cache is to avoid repeatedly
+  // building every cell twice while searching for the last cell in the list.
+  // This cache map structure is "index of cell" => "widget for cell".
+  final Map<int, Widget> _cellWidgetCache = <int, Widget>{};
```

Comment:
Doesn't have to be this PR but moving this to a sliver would keep things small and compositional. Since groups doesn't need any of this stuff so it doesn't have to be in the table.

Reply:
Which parts do groups not need?